inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_gather.cpp

```diff
@@ -66,10 +74,17 @@ void dynamicToStaticShapeGather(std::shared_ptr<ngraph::Node> target) {
         std::iota(second_data_shape_part_indices.begin(), second_data_shape_part_indices.end(), axis + 1);
         const auto second_data_shape_part = std::make_shared<ngraph::opset3::Gather>(
                 data_shape,
-                ngraph::opset3::Constant::create(ngraph::element::i64, {second_data_shape_part_indices.size()}, second_data_shape_part_indices),
-                ngraph::opset3::Constant::create(ngraph::element::i64, {1}, {0}));
+                ngraph::opset3::Constant::create(shapeElementType, {second_data_shape_part_indices.size()}, second_data_shape_part_indices),
+                ngraph::opset3::Constant::create(shapeElementType, {1}, {0}));
         output_dims.push_back(second_data_shape_part);
     }
+    const auto outputShapeIsConst = std::all_of(output_dims.begin(), output_dims.end(), [](const ngraph::Output<ngraph::Node>& output) {
+        return ngraph::as_type_ptr<ngraph::opset3::Constant>(output.get_node_shared_ptr()) != nullptr;
+    });
+    if (outputShapeIsConst) {
+        return;
+    }
```

Comment:
Looks a bit strange. We run this code in case if there is `Gather` with dynamic output. There you're saying there are cases when such a `Gather` is actually static. Could you please explain?

Reply:
We run DTS transformations before CF. So in the beginning there is dynamic input. In cases, when the second input to Gather is the concatenation of constants, it will be folded to constant. But DSR can't pass the shape input as constant: this contradicts the meaning of dynamism. Of course, this check is not sufficient, we cannot detect all cases when shape data will be folded, but it is necessary for many test cases

Reply:
The problem was that in dsr_test_common we set dynamic output shape forcibly and then run DTS.
I added some code that propagates dynamism through the graph and takes output shape from the tested operation. So in cases when gather may resolve dynamism, such output shape will be static and DTS will not be triggered